packages/@react-spectrum/label/src/StaticField.tsx

```diff
@@ -0,0 +1,200 @@
+            <FormattedDateRange value={value as DateTimeValue} formatOptions={formatOptions as Intl.DateTimeFormatOptions} />}
+
+          {typeof value === 'number' && 
+            <FormattedNumber value={value} formatOptions={formatOptions as Intl.NumberFormatOptions} />}
+
+          {(typeof value === 'object' && ('calendar' in value || 'hour' in value) || (value instanceof Date))  &&
+            <FormattedDate value={value} formatOptions={formatOptions as Intl.DateTimeFormatOptions} />}
+
+          {typeof value === 'string' && 
+            props.value}
+        </div>
+      </div>
+    </div>
+  );
+}
+
+function FormattedStringList<T extends string[]>(props: StringListProps<T>) {
+  let stringFormatter = useListFormatter(props.formatOptions); 
+
+  return (
+    <>{stringFormatter.format(props.value)}</>
+  );
+}
+
+function FormattedNumber<T extends NumberValue>(props: NumberProps<T>) {
+  let numberFormatter = useNumberFormatter(props.formatOptions); 
+  let value = props.value;
+
+  if (typeof value === 'object' && typeof (value as RangeValue<NumberValue>).start === 'number' && typeof (value as RangeValue<NumberValue>).end === 'number') {
+    return <>{numberFormatter.formatRange(value.start, value.end)}</>;
+  }
+
+  return <>{numberFormatter.format(value as number)}</>;
+}
+
+function FormattedDateRange<T extends DateTimeValue>(props: DateProps<T>) {
+  let dateFormatter = useDateFormatter(props.formatOptions);
+  let value = props.value;
+  let timeZone = dateFormatter.resolvedOptions().timeZone || getLocalTimeZone();
+  let start;
+  let end;
+
+  if (typeof value === 'object' && 'start' in value) {
+    start = value.start;
+    end = value.end;
+
+    if (!(start instanceof Date) && !(end instanceof Date)) {
+      if (start instanceof Time && end instanceof Time) {
+        start = convertValue(start).toDate(timeZone);
+        end = convertValue(end).toDate(timeZone);
+      } else if (!(start instanceof Time) && !(end instanceof Time)) {
+        start = start.toDate(timeZone);
+        end = end.toDate(timeZone);
+      }
+    } 
+  }
+  
+  return <>{dateFormatter.formatRange(start, end)}</>;
+}
+
+function FormattedDate<T extends DateTimeValue>(props: DateProps<T>) {
+  let dateFormatter = useDateFormatter(props.formatOptions);
+  let value = props.value;
+  let timeZone = dateFormatter.resolvedOptions().timeZone || getLocalTimeZone();
+
+  if (typeof value === 'object' && 'timeZone' in value) {
+    return <>{dateFormatter.format(value.toDate())}</>;
+  }
+
+  if (typeof value === 'object' && 'calendar' in value) {
+    return <>{dateFormatter.format(value.toDate(timeZone))}</>;
+  }
+
+  if (value instanceof Time) {    
+    return <>{dateFormatter.format(convertValue(value).toDate(timeZone))}</>;
+  }
+
+  return <>{dateFormatter.format(value as Date)}</>;
+}
+
+function convertValue(value: TimeValue) {
```

Comment:
I think you only need this if the value is explicitly a `Time`, and not a `CalendarDateTime` or `ZonedDateTime`, which already have a `toDate` method. So the type here could be `Time`, and you could remove some of the logic below (e.g. `day in value`, and the check if it's null).